pvlib/snow.py

```diff
@@ -0,0 +1,182 @@
+"""
+The ``snow`` module contains functions that model the effect of snow on
+solar modules.
+"""
+
+import numpy as np
+import pandas as pd
+from pvlib.tools import sind
+
+
+def _time_delta_in_hours(times):
+    delta = times.to_series().diff()
+    return delta.dt.total_seconds().div(3600)
+
+
+def fully_covered_nrel(snowfall, threshold_snowfall=1.):
+    '''
+    Calculates the timesteps when the row's slant height is fully covered
+    by snow.
+
+    Parameters
+    ----------
+    snowfall : Series
+        Accumulated snowfall in each time period [cm]
+
+    threshold_snowfall : float, default 1.0
+        Hourly snowfall above which snow coverage is set to the row's slant
+        height. [cm/hr]
+
+    Returns
+    ----------
+    boolean: Series
+        True where the snowfall exceeds the defined threshold to fully cover
+        the panel.
+
+    Notes
+    -----
+    Implements the model described in [1]_ with minor improvements in [2]_.
+
+    References
+    ----------
+    .. [1] Marion, B.; Schaefer, R.; Caine, H.; Sanchez, G. (2013).
+       "Measured and modeled photovoltaic system energy losses from snow for
+       Colorado and Wisconsin locations." Solar Energy 97; pp.112-121.
+    .. [2] Ryberg, D; Freeman, J. "Integration, Validation, and Application
+       of a PV Snow Coverage Model in SAM" (2017) NREL Technical Report
+    '''
+    timestep = _time_delta_in_hours(snowfall.index)
+    hourly_snow_rate = snowfall / timestep
+    # if we can infer a time frequency, use first snowfall value
+    # otherwise the first snowfall value is ignored
+    freq = pd.infer_freq(snowfall.index)
+    freq = freq.upper().replace('T', 'M')  # for pandas <= 0.23.4
```

Comment:
might be able to avoid this hack and the `len(freq) == 1` hack by using `to_offset`. For example:

```python
from pandas.tseries.frequencies import to_offset
index = pd.date_range(start="2019-1-1 12:00:00", end="2019-1-1 18:00:00", freq='1h')
offset = to_offset(pd.infer_freq(index))
hr_norm = offset / pd.Timedelta('1h')
```

I've not thoroughly tested against pandas versions.

Reply:
Works for pandas 0.23.4 and later, thanks!

Reply:
FYI I don't think this suggestion made it into the code.

Reply:
It should have, I've got something out of synch between my different computers.